adoc/extensions/sycl_khr_includes.adoc

```diff
@@ -0,0 +1,675 @@
+=== [code]#<sycl/khr/includes/group_algorithms.hpp># header
+
+This header provides definitions of:
+
+* [code]#group_broadcast# function
+* [code]#any_of#, [code]#all_of# and [code]#none_of# functions
+* [code]#shift_left#, [code]#shift_right#, [code]#permute#, [code]#select# and
+  [code]#reduce# functions
+* [code]#exclusive_scan_over_group#, [code]#inclusive_scan_over_group#,
+  [code]#joint_exclusive_scan# and [code]#joint_inclusive_scan# functions
+  functions
+
+[[sec:khr-includes-math]]
+=== [code]#<sycl/khr/includes/math.hpp># header
+
+This header includes all SYCL built-in functions documented by sections:
+
+* <<sec:math-functions>>
+* <<sec:native-precision-math-functions>>
+* <<sec:half-precision-math-functions>>
+* <<sec:integer-functions>>
+* <<sec:common-functions>>
+* <<sec:geometric-functions>>
+* <<sec:relational-functions>>
+
+[[sec:khr-includes-bit]]
+=== [code]#<sycl/khr/includes/bit.hpp># header
+
+This header contains definition of [code]#bit_cast# pre-adopted from the next
+C++ specification.
+
+[[sec:khr-includes-span]]
+=== [code]#<sycl/khr/includes/span.hpp># header
+
+This header contains definitions of [code]#span# and [code]#dynamic_extent#
+pre-adopted from the next C++ specification.
+
+[[sec:khr-includes-byte]]
+=== [code]#<sycl/khr/includes/byte.hpp># header
+
+This header contains definition of [code]#byte# type alias
+
+[[sec:khr-includes-macro]]
+=== Preprocessor directives and macros
+
+In case a SYCL implementation is made using library-only approach, macros
+defined by <<sec:preprocessor-directives-and-macros>> should be available
+regardless of which one of the headers introduced by this extension is included.
+
+For example, [code]#SYCL_LANGUAGE_VERSION# macro should be available even if
+a translation unit within an application only includes
+[code]#<sycl/khr/includes/vec.hpp># header.
+
+[[sec:khr-includes-other-extensions]]
+=== Co-existence with other extensions
+
+The ultimate source of truth about accessibility of an extension through a set
+of fine-grained headers is an extension specification itself. However, there are
+many existing extensions which may not have been updated to account for this
+extension.
+
+Therefore, any extension which does not explicitly document how can it
+be accessed through header files, should be assumed to be available through
+[code]#<sycl/sycl.hpp># only.
+
+{note}It _may_ seem safe to assume that if an extension modifies some existing
+class, then its functionality should be available through a header file which
+exposes that class. However, there exist implementation possibilities where this
+won't be the case and therefore this assumption is not safe to make for
+applications that aim to be portable between SYCL implementations.{endnote}
+
+If an implementation supports this extension together with other extensions,
+then any macro defined by other extensions must be made available through
+[code]#<sycl/khr/includes/version.hpp># header.
```

Comment:
```suggestion
If an implementation supports this extension, then any macro defined by other
supported extensions must be defined in [code]#<sycl/khr/includes/version.hpp>#.
```

Reply:
Applied in cdf04a30b74a521ec25b94a1da55798d2101b156